Backend/src/test/java/controller/internal/DogsControllerTest.java

```diff
@@ -59,4 +60,74 @@ public void AddTest() throws Exception {
                 .andExpect(jsonPath("data.behaviors", hasSize(2)))
                 .andExpect(jsonPath("data.picture.fileName", is("image_name.txt")));
     }
+
+    @Test
+    public void DeleteDogTest() throws Exception {
+
+        mockMvc.perform(
+                MockMvcRequestBuilders.delete("/lostdogs/10001")
+                        .header("token", "regularUserTestToken")
+        ).andExpect(status().isOk())
+                .andExpect(jsonPath("successful", is(true)));
+
+        mockMvc.perform(
+                MockMvcRequestBuilders.delete("/lostdogs/-1")
+                        .header("token", "regularUserTestToken")
+        ).andExpect(status().isBadRequest())
+                .andExpect(jsonPath("successful", is(false)));
+    }
+
+    @Test
+    public void GetDogDetailsTest() throws Exception {
+
+        mockMvc.perform(
+                MockMvcRequestBuilders.get("/lostdogs/10001")
+                        .header("token", "regularUserTestToken")
+        ).andExpect(status().isOk())
+                .andExpect(jsonPath("successful", is(true)));
+
+        mockMvc.perform(
+                MockMvcRequestBuilders.get("/lostdogs/-1")
+                        .header("token", "regularUserTestToken")
+        ).andExpect(status().isBadRequest())
+                .andExpect(jsonPath("successful", is(false)));
+    }
+
+    @Test
+    public void UpdateDogTest() throws Exception {
+
+        MockMultipartFile dog = new MockMultipartFile("dog", "", "application/json", "{\"age\":9,\"breed\":\"random\",\"color\":\"pink\",\"earsType\":\"nie ma\",\"hairLength\":\"long\",\"name\":\"Pinky\",\"size\":\"big\",\"specialMarks\":\"none\",\"tailLength\":\"long\",\"dateLost\":\"2021-03-15\",\"location\":{\"city\":\"Lublin\",\"district\":\"LSM\"},\"behaviors\":[\"Barks loudly\",\"Wags its tail\"]}".getBytes());
+        MockMultipartFile picture = new MockMultipartFile("picture", "image_name.txt", "text/plain", "This should be a picture but we don't have to check for it so it should succeed".getBytes());
+
+        MockMultipartHttpServletRequestBuilder builder = MockMvcRequestBuilders.multipart("/lostdogs/10001");
+        builder.with(new RequestPostProcessor() {
+            @Override
+            public MockHttpServletRequest postProcessRequest(MockHttpServletRequest request) {
+                request.setMethod("PUT");
+                return request;
+            }
+        });
+        mockMvc.perform(builder
+                .file(dog)
+                .file(picture)
+                .header("token", "regularUserTestToken")
+        ).andExpect(status().isOk())
+                .andExpect(jsonPath("successful", is(true)));
```

Comment:
Same applies here, check more stuff than only if code is 200, check if this is actually the dog you wanted to update

Reply:
Here as well.